src/snapred/backend/error/RecoverableException.py

```diff
@@ -0,0 +1,31 @@
+from typing import Any, Literal
+
+from snapred.backend.log.logger import snapredLogger
+
+logger = snapredLogger.getLogger(__name__)
+
+RecoverableErrorType = Literal[
+    "AttributeError: 'NoneType' object has no attribute 'instrumentState'",
+    "etc",
+]
+
+
+class RecoverableException(Exception):
+    """
+    Raised when a recoverable exception is thrown.
+    Allows for custom error types and optional extra context.
+    """
+
+    def __init__(self, exception: Exception, errorType: RecoverableErrorType, **kwargs: Any):
```

Comment:
Switched to `errorMsg`.

'calibrationState is None' because from within `LocalDataService` the method used to check state checks based on a path that is constructed with a set of hex values (stateID). The returned variable remains `None` if this path does not exist (e.i. os.path.exists(calibrationStatePath)). It does not return the error ['"AttributeError: 'NoneType' object has no attribute 'instrumentState'"] until later on in the execution.